action.yml

```diff
@@ -47,7 +47,7 @@ runs:
       run: |
         mkdir ~/osv-detector && cd ~/osv-detector
         echo "Installing osv-detector v$OSVDETECTOR_VERSION ($OSVDETECTOR_BINARY)"
-        curl -fL https://github.com/G-Rath/osv-detector/releases/download/v$OSVDETECTOR_VERSION/$OSVDETECTOR_BINARY --output osv-detector
+        curl --fail --connect-timeout 5 --max-time 5 --retry-delay 0 --retry-max-time 30 --retry-all-errors -fL https://github.com/G-Rath/osv-detector/releases/download/v$OSVDETECTOR_VERSION/$OSVDETECTOR_BINARY --output osv-detector
```

Comment:
@scalp42 are --fail` and `--retry-all-errors` necessary? `--retry` will retry any 5xx errors, so if `--fail` causes 4xx to be treated as if it was a 5xx, I would assume that's sufficient for the normal retry mechanism to be triggered?

Thanks for this by the way. @G-Rath and I have been going back and forth on Slack trying to figure out how to nicely retry accessing the GH release URL :-)

Reply:
Fair enough, I'll remove `--fail` and we'll rely on `--retry-all-errors` sledgehammer. Commit inc.

Reply:
for extra context, one of the reasons this can fail is because of rate limiting since we're hitting a GH API endpoint (the release download in the next step doesn't seem to be subject to rate limiting though) - I'm wondering if we should at least ensure that the retrying goes for up to 60 seconds since that's how long it takes for this rate limit to reset (though 30 seconds might be good enough too 🤔)

Reply:
I'm happy with whatever, we've never seen any rate-limiting errors on our end though.

Just let me know.

Reply:
The rate limiting is in the headers, which you won't see here unless you're explicitly looking for them via a dedicated `cURL` because nothing is outputted here; but I'm happy with this, I just wanted to share that context incase either of us continue to see some failures in which case bumping the retry number here is probably a good first step

Reply:
Gotcha, I'm a bit confused as I thought it'd return 403s 😅